File: lab1 class/untitled/src/Serum.java
public class Serum extends SkinCare {
    private boolean hyppoallergenic;
    private String speedOfAction;
    protected Serum(String name, String brand,  double price, String activeComponent, String scent, boolean hyppoallergenic, String speedOfAction ) {
        super (name, brand,price, activeComponent, scent);
        this.hyppoallergenic = hyppoallergenic;
        this.speedOfAction = speedOfAction;
    }


    @Override
    public void printinfo() {
        System.out.println(name);
        System.out.println("Brand: "+ brand);
        System.out.println("Price: "+ price + "$ ");
        System.out.println("Speed of action: " + speedOfAction);
        System.out.println("Scent: " + scent);
        if (hyppoallergenic ){
            System.out.println("Hyppoallergetic \n");}
        else {
            System.out.println("");
        }
    }
}

File: lab2_fabrica/fabrica/Serum.cs
using System;
using System.Drawing;
using System.Runtime.Serialization;


namespace cosmetics
{
    [DataContract]   

    
    internal class Serum : SkinCare
    {
        [DataMember]
        private Boolean hyppoallergenic;
        [DataMember]
        private String speedOfAction;
        public Serum(String name, String brand, double price, String activeComponent, Boolean hyppoallergenic, String speedOfAction):base (name, brand, price, activeComponent)
        {
            this.hyppoallergenic = hyppoallergenic;
            this.speedOfAction = speedOfAction;
        }
        public Boolean Hyppoallergenic
        {
            set => hyppoallergenic = value;
            get => hyppoallergenic;
        }
        public String SpeedOfAction
        {
            get => speedOfAction;
            set => speedOfAction = value;
        }

        public override String getinfo()
        {
            String info =
            "Name: " + name + "\r\n" +
            "Brand: " + brand + "\r\n" +
            "Price: " + price + "$" + "\r\n" +
            "Active component: " + activeComponent+ "\r\n";
            if (hyppoallergenic)
            {
                info += "Hyppoallergetic  \r\n";
            }

            info += "Speed of action: " + speedOfAction + "\r\n";
            return info;
        }

        public override Bitmap Img()
        {
            return (Bitmap)Bitmap.FromFile(".//img//serum.png");
        }

    }
}
